src/Api/SecretsManager/Controllers/SecretsController.cs

```diff
@@ -87,18 +86,23 @@ public async Task<SecretResponseModel> GetAsync([FromRoute] Guid id)
             throw new NotFoundException();
         }
 
-        if (!await UserHasReadAccessToSecret(secret))
+        var userId = _userService.GetProperUserId(User).Value;
+        var orgAdmin = await _currentContext.OrganizationAdmin(secret.OrganizationId);
+        var accessClient = AccessClientHelper.ToAccessClient(_currentContext.ClientType, orgAdmin);
+
```

Comment:
With our approach, we will end up with something like this in most of the SM controllers.

Is there somewhere we could put something like this to DRY up the pattern?

```
    private async Task<(AccessClientType AccessClientType, Guid UserId)> GetAccessClientTypeAsync(Guid organizationId)
    {
        if (!_currentContext.AccessSecretsManager(organizationId))
        {
            throw new NotFoundException();
        }

        var userId = _userService.GetProperUserId(User).Value;
        var orgAdmin = await _currentContext.OrganizationAdmin(organizationId);
        var accessClient = AccessClientHelper.ToAccessClient(_currentContext.ClientType, orgAdmin);
        return (accessClient, userId);
    }
```

I don't think this is required for this specific PR.

Reply:
Seems reasonable, not in scope for this PR though.